libbeat/esleg/eslegclient/connection.go

```diff
@@ -311,16 +314,113 @@ func (conn *Connection) Close() error {
 	return nil
 }
 
+type httpClientProxySettings struct {
+	*httpcommon.HTTPClientProxySettings
+}
+
+// ProxyDialer is a dialer that can be registered to golang.org/x/net/proxy
+func (settings *httpClientProxySettings) ProxyDialer(_ *url.URL, forward proxy.Dialer) (proxy.Dialer, error) {
+	return transport.DialerFunc(func(_, address string) (net.Conn, error) {
+
+		// Headers given to the CONNECT request
+		hdr := settings.Headers.Headers()
+		if settings.URL.User != nil {
+			username := settings.URL.User.Username()
+			password, _ := settings.URL.User.Password()
+			if len(hdr) == 0 {
+				hdr = http.Header{}
+			}
+			hdr.Add("Proxy-Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(username+":"+password)))
+		}
+
+		req := &http.Request{
+			Method: "CONNECT",
+			URL:    &url.URL{Opaque: address},
+			Host:   address,
+			Header: hdr,
+		}
+
+		// Dial the proxy host
+		c, err := forward.Dial("tcp", settings.URL.Host)
+		if err != nil {
+			return nil, err
+		}
+
+		// Write the CONNECT request
+		err = req.Write(c)
+		if err != nil {
+			c.Close()
+			return nil, err
+		}
```

Comment:
It's just a detail, but you can be a bit more compact here:
```suggestion
		// Write the CONNECT request
		if err := req.Write(c); err != nil {
			c.Close()
			return nil, err
		}

```